code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -103,14 +103,14 @@
 	add_overlay(base_dead_overlay)
 
 /mob/living/simple_animal/hostile/carp/proc/chomp_plastic()
-	var/obj/item/storage/cans/tasty_plastic = locate(/obj/item/storage/cans) in oview(src, 9)
+	var/obj/item/storage/cans/tasty_plastic = locate(/obj/item/storage/cans) in view(1, src)
 	if(tasty_plastic)

```

Comment:
```suggestion
	if(tasty_plastic && Adjacent(tasty_plastic))
```
Could be behind a directional window or windoor.
Bonus points if you do an early return instead.